service/worker/scheduler/workflow.go

```diff
@@ -636,18 +716,32 @@ func (s *scheduler) processSignals() bool {
 	return scheduleChanged
 }
 
-func (s *scheduler) getFutureActionTimes(n int) []*time.Time {
-	// Note that `s` may be a fake scheduler used to compute list info at creation time.
+func (s *scheduler) getFutureActionTimes(inWorkflowContext bool, n int) []*time.Time {
+	// Note that `s` may be a fake scheduler used to compute list info at creation time
```

Comment:
nit: I'd find another way to describe this without using "fake" because that sounds like a test object

Reply:
sure.

btw, instead of the `bool` argument, another way to do this would be to check `s.ctx != nil`. I figured the bool was cleaner but the other also makes some sense.